src/Morphir/Visual/Components/Theme.elm

```diff
@@ -0,0 +1,24 @@
+module Morphir.Visual.Components.Theme exposing (..)
+
+import Element
+
+
+type alias Theme =
+    { fontSize : Int
+    , smallSpacing : Int
+    , mediumSpacing : Int
+    , largeSpacing : Int
+    , smallPadding : Int
+    , mediumPadding : Int
+    , largePadding : Int
```

Comment:
Can you change these into functions of the form:

```elm
smallSpacing : Theme -> Int
smallSpacing theme =
    scaled -3 theme.fontSize
```

This way you can include specific values like `-3` in the `Theme` itself which is where they belong. You don't want different behavior depending on how the client of the `Theme` (`DevelopApp`, `Insight`) decides to define these.

Reply:
But  It will look really bad when I use them in rows and columns. Ex- Element.row [ spacing (config.smallSpacing theme) ] []

Reply:
I think it will be:

```elm
row [ spacing (theme |> Theme.smallSpacing) ] []
```

Which doesn't look bad in my opinion. You can also simplify it by returning an actual `Element.Attribute` so that you can do just:

```elm
row [ theme |> Theme.smallSpacing ] []
```

Reply:
Thinking about it a bit more I realized that the main issue I had was with the fact that the creation of the `Theme` record was not centralized. As long as you can move that into the `Theme` module itself you don't need to change the fields into functions. The main point is that the scaling ratios should be managed centrally instead of user code.

Reply:
I understand what you are saying.  So I think  `row [ theme |> Theme.smallSpacing` will be a good way to manage.

Reply:
I got some problem with this approach when we have some custom padding like `paddingXY 12 14` and `noPadding { left = .. , right = .. , up = .. , down = .. }` . I am not able to update these with  approach `row [ theme |> Theme.smallSpacing ] []` because they require Int am I am returning Element.attribute msg.